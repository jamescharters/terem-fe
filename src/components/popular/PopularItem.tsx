import React from "react";
import Styles from "./PopularItem.module.scss";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faLocationPin } from "@fortawesome/free-solid-svg-icons";
import { TileItem } from "@/types/TileItem";

export const PopularItem: React.FC<TileItem> = (props) => {
    return (
        <div className={Styles.container}>
            <img alt={props.title} src={props.imageUrl} />
            <div>
                <h4>{props.title}</h4>
                <p>
                    <FontAwesomeIcon icon={faLocationPin} />
                    <span>{props.location}</span>
                </p>
            </div>
        </div>
    );
};

export default PopularItem;
