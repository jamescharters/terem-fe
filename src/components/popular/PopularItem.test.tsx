import { render } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import PopularItem from "./PopularItem";

describe("PopularItem", () => {
    it("renders the component", () => {
        const result = render(<PopularItem imageUrl="http://a.b" linkUrl="http://b.c" location="Sydney, Australia" title="A Fake Title" />);
        expect(result.asFragment()).toMatchSnapshot();
    });
});
