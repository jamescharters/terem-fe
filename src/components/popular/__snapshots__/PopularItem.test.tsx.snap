// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PopularItem > renders the component 1`] = `
<DocumentFragment>
  <div
    class="_container_0cedfc"
  >
    <img
      alt="A Fake Title"
      src="http://a.b"
    />
    <div>
      <h4>
        A Fake Title
      </h4>
      <p>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-location-pin "
          data-icon="location-pin"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M384 192c0 87.4-117 243-168.3 307.2c-12.3 15.3-35.1 15.3-47.4 0C117 435 0 279.4 0 192C0 86 86 0 192 0S384 86 384 192z"
            fill="currentColor"
          />
        </svg>
        <span>
          Sydney, Australia
        </span>
      </p>
    </div>
  </div>
</DocumentFragment>
`;
